tests/bson/typemap-003.phpt

```diff
@@ -0,0 +1,81 @@
+--TEST--
+MongoDB\Driver\Cursor::setTypeMap(): Setting and replacing typemaps
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php NEEDS('STANDALONE'); CLEANUP(STANDALONE); ?>
+--FILE--
+<?php
+
+require_once __DIR__ . "/../utils/basic.inc";
+
+class MyArrayObject extends ArrayObject implements MongoDB\BSON\Unserializable
+{
+    function bsonUnserialize(array $data)
+    {
+        parent::__construct($data, \ArrayObject::ARRAY_AS_PROPS);
+    }
+}
+
+class MyProperties extends MyArrayObject
+{
+}
+
+$manager = new MongoDB\Driver\Manager(STANDALONE);
+
+$document1 = [
+    '_id' => 1,
+    'array' => [1, 2, 3],
+    'object' => ['string' => ['sleutels', 'keys'] ]
+];
+$document2 = [
+    '_id' => 2,
+    'array' => [4, 5, 6],
+    'object' => ['associative' => ['elementen', 'elements' ]]
```

Comment:
This test was originally to see if we could set the typemap twice without leaking or memory issues.

But I added a functional bit to test that a bit better, and might ended up having just a test for PROPS_AS_ARRAY. Because of the memory issue/doubble setting, I would like to keep it though.

Reply:
I'm still not sure how `ARRAY_AS_PROPS` relates to the double setting of a type map, but no objections if you'd like to keep this in place.

Reply:
It doesn't — ``ARRAY_AS_PROPS`` was just added to make the asserts a little easier to write.